File: Architecture/TodoApp/android/mobile/src/main/java/com/novoda/todoapp/BaseActivity.java
package com.novoda.todoapp;

import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void setContentView(@LayoutRes int layoutResID) {
        super.setContentView(layoutResID);
        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
    }

}

File: TestAutomationStarter/app/src/main/java/com/novoda/androidstoreexample/activities/BaseActivity.kt
package com.novoda.androidstoreexample.activities

import android.app.ProgressDialog
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.novoda.androidstoreexample.dagger.App
import com.novoda.androidstoreexample.dagger.component.AppComponent
import com.novoda.androidstoreexample.mvp.view.BaseView

abstract class BaseActivity: AppCompatActivity(), BaseView {

    abstract fun getActivityLayout(): Int

    abstract fun injectDependencies(appComponent: AppComponent)

    private var progressDialog: ProgressDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getActivityLayout())
        injectDependencies(App.component)
        initProgressDialog()
    }

    private fun initProgressDialog() {
        progressDialog = ProgressDialog(this)
        progressDialog?.setProgressStyle(ProgressDialog.STYLE_SPINNER)
        progressDialog?.setMessage("Please Wait")
    }

    override fun showProgress() {
        progressDialog?.show()
    }

    override fun hideProgress() {
        progressDialog?.hide()
    }

    override fun showMessage(message: String) {

    }
}